application/src/main/java/dev/aws101/todo/TodoServiceImpl.java

```diff
@@ -1,35 +1,52 @@
 package dev.aws101.todo;
 
 import dev.aws101.collaboration.TodoCollaborationRequest;
+import dev.aws101.collaboration.TodoCollaborationRequestRepository;
 import dev.aws101.person.Person;
 import dev.aws101.person.PersonRepository;
+import org.apache.commons.codec.digest.DigestUtils;
 import org.slf4j.Logger;
 import org.slf4j.LoggerFactory;
 import org.springframework.beans.factory.annotation.Value;
+import org.springframework.cloud.aws.messaging.core.NotificationMessagingTemplate;
 import org.springframework.cloud.aws.messaging.core.QueueMessagingTemplate;
 import org.springframework.security.core.context.SecurityContextHolder;
 import org.springframework.security.oauth2.core.oidc.user.DefaultOidcUser;
 import org.springframework.stereotype.Service;
 
+import java.time.LocalDateTime;
+
 @Service
 public class TodoServiceImpl implements TodoService {
```

Comment:
this class now really grows in responsibility. I guess we can refactor this later on to smaller components.

Reply:
The collaboration-related methods could be moved to a new service. The `testConfirmCollaboration` could be moved to a service that's only available for the dev profile.

Reply:
Done.